src/pages/scripting/how-it-works/index.md

```diff
@@ -0,0 +1,100 @@
+---
+title: How it works
+description: More details about UXP Scripting
+contributors:
+  - https://github.com/amandahuarng
+---
+
+# How UXP Scripting Works
+Each Photoshop script file is a plain text file with a `.psjs` file extension. We plan on expanding UXP scripting to other applications and supporting opening scripts via double-click in the future. When this happens, the file extension will help identify which application the script should be launched in. 
+
+## Execution Context
+Photoshop sets an execution context while invoking a script.
+
+Within each execution context, only one script can be executed at a time. You cannot invoke another script from the running script. Using the UXP script module, you can access `ExecutionContext`.
+
+```js
+const script = await require("uxp").script;
+const executionContext = script.executionContext;
+```
+It provides details about the current script execution, methods to send data to Photoshop, and events to manage script lifecycles. You can also use methods belonging to `executionControl.hostControl` to suspend/resume history states and auto close documents. Read more in the scripting [reference](../reference/).
+
+## Events
+Script execution can be cancelled when: 
+* a user clicks on "Cancel" in the progress bar
+* Photoshop encounters some exception in running a script file 
+
+Script developers can add event handlers to get notified when the command has been cancelled. The callback will receive "reason" as a parameter. 
+
+```js
+executionContext.onCancel.addListener((reason) => {
+    // reason would be a json object set by PS while cancelling
+    reject("Message");
+});
+``` 
+
+## User Interface
+Scripts can only show a dialog UI. Any UI created by a script is modal in nature and must use **global await** or it will be destroyed when the script is done running. Photoshop automatically shows a **progress bar** if the script takes more than 2-3 seconds to finish.
```

Comment:
"must use global await"
I think this is misleading.  Any await is sufficient for a showModal.  The point is that if you allow execution to continue (by not using await somewhere encompassing the showModal letting it return a Promise), execution _can_ proceed to completion.  At which point, the UI is destroyed.  If you blink, you might miss it.

Perhaps a cautionary note would be advisable here to say something akin to, "If you don't see your modal UI appear, check that there is an `await` in front of it."

Reply:
Changed it to be: "Scripts can only show a dialog UI. Any UI created by a script is modal in naturel By *not* using `await` for `showModal()` (not letting it return a Promise), execution can continue to completion at which point any UI is destroyed. If you don't see your modal UI appear, check that there is an `await` in front of it."